Modules/_ssl.c

```diff
@@ -187,6 +187,11 @@ extern const SSL_METHOD *TLSv1_2_method(void);
 #endif
 
 
+#if !defined(SSL_VERIFY_POST_HANDSHAKE) || !defined(TLS1_3_VERSION) || defined(OPENSSL_NO_TLS1_3)
+  #define PY_SSL_NO_POST_HS_AUTH
```

Comment:
I'd prefer a positive name: it is always used with `!` later and double negatives are harder to read.

Perhaps `PySSL_HAVE_POST_HS_AUTH`